cfn-resources/cloud-backup-snapshot/cmd/resource/resource.go

```diff
@@ -244,23 +216,32 @@ func isSnapshotExist(currentModel *Model) bool {
 		return false
 	}
 
-	projectID := *currentModel.GroupId
+	// Create Atlas API Request Object
+	clusterName := cast.ToString(currentModel.ClusterName)
+	instanceName := cast.ToString(currentModel.InstanceName)
+	projectID := cast.ToString(currentModel.GroupId)
 	snapshotRequest := &mongodbatlas.SnapshotReqPathParameters{
-		GroupID:     projectID,
-		ClusterName: *currentModel.ClusterName,
+		GroupID: projectID,
 	}
-
+	var snapshots *mongodbatlas.CloudProviderSnapshots
 	params := &mongodbatlas.ListOptions{
 		PageNum:      0,
 		ItemsPerPage: 100,
 	}
 
-	snapshots, _, err := client.CloudProviderSnapshots.GetAllCloudProviderSnapshots(context.Background(), snapshotRequest, params)
+	// API call to list snapshot
+	if clusterName != "" {
+		snapshotRequest.ClusterName = clusterName
+		snapshots, _, err = client.CloudProviderSnapshots.GetAllCloudProviderSnapshots(context.Background(), snapshotRequest, params)
+	} else {
+		// read serverless instance snapshot
+		snapshotRequest.InstanceName = instanceName
```

Comment:
similarly to above with `clusterName`, should we have some check to be sure that `instanceName` is populated?

Reply:
fixed this comment